src/button-group/__tests__/button-group-dev.test.tsx

```diff
@@ -0,0 +1,140 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
I reworked button group unit tests. Now all tests are split across multiple files focusing on specific categories of tests. The tests ensure full coverage for all button group item types.

In the dev tests we validate dev warnings, fallbacks, and callbacks.